src/recovery/logical.c

```diff
@@ -809,11 +857,63 @@ orioledb_decode(LogicalDecodingContext *ctx, XLogRecordBuffer *buf)
 		else if (rec_type == WAL_REC_ROLLBACK_TO_SAVEPOINT)
 		{
 			SubTransactionId parentSubid;
+			OXid		xmin;
+			dlist_iter	cur_txn_i;
+			ReorderBufferTXN *txn;
+			CommitSeqNo csn;
+			CSNSnapshotData *csnSnapshot;
 
-			memcpy(&parentSubid, ptr, sizeof(SubTransactionId));
-			ptr += sizeof(SubTransactionId);
+			xmin = InvalidOXid;
+			csn = 0;
 
+			ptr = wal_parse_rec_rollback_to_savepoint(ptr, &parentSubid, &xmin, &csn);
+
+#if ORIOLEDB_WAL_VERSION < (17)
 			/* Skip */
+			continue;
+#endif
+
```

Comment:
```suggestion
```

Reply:
Change this to runtime check `if (wal_version < 17)` - then skip and continue